Core/Loaders.cpp

```diff
@@ -336,9 +338,20 @@ bool LoadFile(FileLoader **fileLoaderPtr, std::string *error_string) {
 	case IdentifiedFileType::UNKNOWN_BIN:
 	case IdentifiedFileType::UNKNOWN_ELF:
 	case IdentifiedFileType::UNKNOWN:
+		ERROR_LOG(LOADER, "Unknown file type: %s (%s)", fileLoader->GetPath().c_str(), error_string->c_str());
+		*error_string = "Unknown file type: " + fileLoader->GetPath().ToString();
+		break;
+
+	case IdentifiedFileType::ERROR_IDENTIFYING:
+		*error_string = *error_string + ": " + (fileLoader ? fileLoader->LatestError() : "");
+		if (error_string->empty())
+			*error_string = "Error reading file";
```

Comment:
Hm, this isn't possible, is it?  Should this be above the previous line?

-[Unknown]

Reply:
Yeah, kinda screwed up moving things around there. I'll fix it up.